### PR TITLE
179597784 completion page

### PR DESCRIPTION
**Pull request by eireland**
When user sets page to be a completion page in page settings:
- Shows an authoring Completion Page (using existing authoring Completion Page text) 
- Moves that page to the end of the pages array so that it is now the last page of the activity.
- When a new page is added, and there is a completion page, the new page is added to the page prior to the completion page
Note that if the author changes the setting so it is no longer the completion page, that page remains at the end.

### CODE REVIEW

**Review by emcelroy — COMMENTED** [review]
I think this looks good and it works perfectly well when I test it locally. There are two things I'm not sure of, though, and think we should wait till @knowuh takes a look before merging.

I'll make separate comments inline with the parts I'm not sure about.

File: app/controllers/api/v1/interactive_pages_controller.rb
Comment (emcelroy) [inline app/controllers/api/v1/interactive_pages_controller.rb:66]: As I understand it `move_lower` was deprecated as of Rails 2. I'm not sure it matters, but it may be better to find a different way to do this.

File: lara-typescript/src/section-authoring/api/lara-api-provider.ts
Comment (emcelroy) [inline lara-typescript/src/section-authoring/api/lara-api-provider.ts:103]: I'm not sure this matters, either, but might it be better to have the Rails side handle all of this?

File: app/controllers/api/v1/interactive_pages_controller.rb
Comment (eireland) [reply app/controllers/api/v1/interactive_pages_controller.rb:66]: I got it from here? https://github.com/brendon/acts_as_list

Comment (emcelroy) [reply app/controllers/api/v1/interactive_pages_controller.rb:66]: Hm. Maybe it's not the same thing I was looking at then which is [this](https://apidock.com/rails/ActiveRecord/Acts/List/InstanceMethods/move_lower). I see you're discussing acts_as_list with Noah in Slack, so I'll defer to the outcome of that conversation about this.

File: app/controllers/api/v1/interactive_pages_controller.rb
Comment (emcelroy) [inline app/controllers/api/v1/interactive_pages_controller.rb:93]: Seems like the comma at the end of this line isn't necessary. Should it be removed?

**Review by emcelroy — APPROVED** [review]
I left a few comments about possibly cleaning up some minor bits of unused or unnecessary code, but otherwise this looks good and works as expected when tested locally. Thanks.

**Pull request merged**
